snapcraft_legacy/internal/pluginhandler/_part_environment.py

```diff
@@ -84,6 +86,12 @@ def get_snapcraft_global_environment(
         "SNAPCRAFT_TARGET_ARCH": project.target_arch,
         "SNAPCRAFT_CONTENT_DIRS": content_dirs_envvar,
     }
+    # run-on architecture may not be available
```

Comment:
if the run on architecture is not available, then we source the build_on as the convention is that they should match, and perhaps this is the culprit to all problems

Reply:
calling it build_for makes the intent a bit more obvious too

Reply:
I changed `run-on` to `build-for` and reworded this comment